source/extensions/network/dns_resolver/apple/apple_dns_impl.h

```diff
@@ -119,6 +119,8 @@ class AppleDnsResolverImpl : public DnsResolver, protected Logger::Loggable<Logg
     // onDNSServiceGetAddrInfoReply callback.
     struct PendingResponse {
       ResolutionStatus status_;
+      bool v4_response_received_;
```

Comment:
nit: Comments on the new members. In particular, for these two it would probably be nice to explain the semantics. Should this be true if, say, the v4 query returned NXDOMAIN (no entities) or an error?

Consider adding a member initializer for the POD fields which would make the default constructor do something useful.